byway/src/main/webapp/recommendations.js

```diff
@@ -12,102 +12,65 @@
+/* exported calcMainRoute, recs */
 
-if(document.readyState === 'loading') {
-  document.addEventListener('DOMContentLoaded', initServices);
-} else {
-  initServices();
-}
-
-let map;
-let placesService;
+// Holds recommendations as PlaceResult objects
+let recs = [];
 
 // Measured in meters
 const RADIUS_TO_SEARCH_AROUND = 1000;
 const MIN_DISTANCE_FOR_STEP_PATH = 4000;
 
-const interests = ["park"];
-let destinations = new Set();
+// Max recommendations per interest
+const MAX_RECOMMENDATIONS = 1;
 
 // Contains google.maps.LatLng objects.
 // Used as a center point to search around a region.
 let regions = [];
 
 /**
- * Initializes the webpage with a map and other google
- * services. Creates a route between two endpoints.
- * Note: Temp setup to mimic a route between two endpoints.
- * TODO: Link/adapt with Justine's third page, which handles
- * initializing the map and other google services.
- */
-function initServices() {
-  // Modified version of Justine's implementation
-  let directionsService = new google.maps.DirectionsService();
-  let directionsRenderer = new google.maps.DirectionsRenderer();
-  const start = new google.maps.LatLng(37.7699298, -122.4469157);
-  const end = new google.maps.LatLng(37.7683909618184, -122.51089453697205);
-  const mapOptions = {
-    zoom: 14,
-    center: start
-  }
-  addDestinations(start, end);
-  map = new google.maps.Map(document.getElementById('map'), mapOptions);
-  directionsRenderer.setMap(map);
-  document.getElementById("route").addEventListener("click", () => {
-    calcRoute(directionsService, directionsRenderer, start, end);
-  });
-  placesService = new google.maps.places.PlacesService(map);
-}
-
-/**
- * Add start and end points of the route to regions to search around.
- * Temporary set up to mimic adding several endpoints between legs.
- * @param {google.maps.LatLng} start coordinate of route's startpoint
- * @param {google.maps.LatLng} end coordinate of route's endpoint
- */
-function addDestinations(start, end) {
-  destinations.add(start);
-  destinations.add(end);
-  regions.push(...destinations);
-}
-
-/**
- * Creates a route between two points and loads onto the map.
- * Finds points along the path to load the regions Array.
- * Loads recommendations centered around these points.
- * @param {DirectionsService} directionsService finds directions
- * @param {DirectionsRenderer} directionsRenderer renders the route
- * @param {LatLng} start starting point location
- * @param {LatLng} end ending point location
+ * Creates round-trip route with waypoints that loads onto the map.
+ * Partitions the route into regions, used to load recommendations
+ * around. Orders the waypoints for efficiency and updates trip logistics.
  */
-function calcRoute(directionsService, directionsRenderer, start, end) {
+function calcMainRoute() {
   resetUserAlerts();
   const request = {
-      origin:  start,
-      destination: end,
-      travelMode: 'DRIVING'
+    origin:  start.name,
```

Comment:
where do start and end variables come from? i dont see them declared globally or passed into the function

Reply:
They are declared in routepage.js, and used in recommendations.js. They are passed in globally, and there is also a comment with `/*global name*/` for the eslint. (https://github.com/googleinterns/step200-2020/pull/38/files/9e355c6dbbcc19a0ec2597b8dc7d8a4b22840b39#diff-0ab10c2304e3f9a2628ceb4c849183cbR15-R16)